sdk/eventhubs/azure-messaging-eventhubs/inc/azure/messaging/eventhubs/checkpoint_store.hpp

```diff
@@ -122,18 +103,23 @@ namespace Azure { namespace Messaging { namespace EventHubs {
      *
      * @param connectionString  The connection string of an Azure Storage account.
      * @param containerName  The name of the blob container.
+     * @param blobClientOptions options for the underlying blob client.
      */
-    BlobCheckpointStore(std::string const& connectionString, std::string const& containerName)
+    BlobCheckpointStore(
+        std::string const& connectionString,
+        std::string const& containerName,
+        Azure::Storage::Blobs::BlobClientOptions blobClientOptions = {})
```

Comment:
Rather than creating a client for the user, should we instead accept a `BlobContainerClient` instance? This way, we don't have to worry about extending this for the various ways we may create a blob container client, in the future.

Reply:
That's actually a *really* good idea.

Done.